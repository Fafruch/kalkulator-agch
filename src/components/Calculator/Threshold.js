import React from 'react'
import PropTypes from 'prop-types'

const Threshold = ({ courseScore, threshold }) => (
  <div className={courseScore >= threshold
    ? 'score-above'
    : 'score-under'}
  >
    {threshold}
  </div>
)

Threshold.propTypes = {
  courseScore: PropTypes.number.isRequired,
  threshold: PropTypes.number
}

export default Threshold
