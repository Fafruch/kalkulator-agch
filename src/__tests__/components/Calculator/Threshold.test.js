import React from 'react'
import { shallow } from 'enzyme'

import Threshold from '../../../components/Calculator/Threshold'

const setup = (setupProps = {}) => {
  const defaultProps = {
    courseScore: 920,
    threshold: 914
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<Threshold {...props} />)

  return {
    props,
    wrapper
  }
}

describe('Threshold component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  test('should has proper class depending on the threshold value', () => {
    let { wrapper } = setup()

    expect(wrapper.is('.score-above')).toEqual(true)

    ;({ wrapper } = setup({ courseScore: 920, threshold: 1000 }))

    expect(wrapper.is('.score-under')).toEqual(true)
  })
})
