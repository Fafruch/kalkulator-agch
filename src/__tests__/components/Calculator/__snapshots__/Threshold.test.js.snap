// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Threshold component should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="score-above"
>
    914
</div>,
  "nodes": Array [
    <div
      className="score-above"
>
      914
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Threshold
        courseScore={920}
        threshold={914}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "courseScore": 920,
          "threshold": 914,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="score-above"
>
          914
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="score-above"
>
          914
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Threshold
    courseScore={920}
    threshold={914}
/>,
}
`;
